packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -587,9 +590,12 @@ class OffsetLayer extends ContainerLayer {
   Future<ui.Image> toImage(Rect bounds, {double pixelRatio = 1.0}) async {
     assert(bounds != null);
     assert(pixelRatio != null);
+    final Matrix4 transform = new Matrix4.translationValues(
```

Comment:
I now better understand how offset vs transform is used and to be robust the offset and bounds need to be rolled into the transform here instead of passed down in the offset. Otherwise there are bugs in CompositedTransformFollower and other cases where the offset is intentionally ignored.

Reply:
Good news is this new logic works and has an explicit test in the rendering layer as well as implicit tests at the widget layer for all the screenshot tests inf widget_inspector_test